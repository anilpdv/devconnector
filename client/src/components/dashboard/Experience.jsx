import React, { Component } from "react";
import { connect } from "react-redux";
import PropTypes from "prop-types";

export class Experience extends Component {
  render() {
    const experience = this.props.experience.map(exp => (
      <tr className="" key={exp._id}>
        <td className="row">{exp.company}</td>
        <td className="row">{exp.title}</td>
        <td className="row">
          {exp.from} - {exp.to}
        </td>
        <td>
          <button className="btn btn-danger">delete</button>
        </td>
      </tr>
    ));
    return (
      <div>
        <table className="table">
          <thead>
            <tr>
              <th className="col">company</th>
              <th className="col">title</th>
              <th className="col">years</th>
            </tr>
          </thead>
          <tbody>
            <tr>{experience}</tr>
          </tbody>
        </table>
      </div>
    );
  }
}

Experience.propTypes = {
  experience: PropTypes.object.isRequired
};

export default connect(null)(Experience);
