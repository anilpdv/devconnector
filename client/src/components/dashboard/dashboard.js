import React, { Component, Fragment } from "react";
import { connect } from "react-redux";
import Loader from "react-loader-spinner";
import PropTypes from "prop-types";
import { getCurrentProfile, deleteProfile } from "../../actions/profile";
import { setErrorsEmpty, setLoggedIn } from "../../actions/auth";
import { Link } from "react-router-dom";
import ProfileActions from "./ProfileActions";
import { toast, ToastContainer } from "react-toastify";
import "react-toastify/dist/ReactToastify.css";

export class Dashboard extends Component {
  componentDidMount() {
    if (this.props.getCurrentProfile) {
      this.props.getCurrentProfile();
    }

    if (this.props.setErrorsEmpty) {
      this.props.setErrorsEmpty();
    }

    if (this.props.auth.loggedIn) {
      this.alertSuccess();
      this.props.setLoggedIn(false);
    }
  }

  alertSuccess = () =>
    toast.success("Successfully logged in", {
      position: "top-right",
      autoClose: 3000,
      hideProgressBar: false,
      closeOnClick: true,
      pauseOnHover: true,
      draggable: true
    });

  render() {
    const { user } = this.props.auth ? this.props.auth : { user: "" };

    const { profile, loading } = this.props.profile
      ? this.props.profile
      : { profile: "", loading: "" };

    let dashboardContent;
    if (profile === null || loading) {
      console.log({ profile });
      console.log({ loading });
      dashboardContent = (
        <Loader
          type="ThreeDots"
          style={{ position: "fixed", top: "45%", left: "50%" }}
          color="#989898"
          height={40}
          width={40}
        />
      );
    } else {
      if (Object.keys(profile).length > 0) {
        dashboardContent = (
          <Fragment>
            <p>
              welcome to
              <Link to={`/profile/${profile.profile.handle}`}>
                {" "}
                {user.name}{" "}
              </Link>
            </p>
            <ProfileActions />
            {/* TODO : education and profile */}
            <div className="delete">
              <button
                className="btn btn-danger"
                onClick={() => {
                  this.props.deleteProfile();
                }}
              >
                Delete
              </button>
            </div>
          </Fragment>
        );
      } else {
        dashboardContent = (
          <div>
            <p className="">welcome {user ? user.name : ""}</p>
            <span className="text-muted pr-4">
              we cannot find profile, create a profile here!
            </span>
            <Link className="btn btn-success" to="/create-profile">
              create Profile
            </Link>
          </div>
        );
      }
    }

    return (
      <div className="Dashboard">
        <div className="container">
          <div className="row">
            <div className="col-md-12">
              <div className="dispaly-4">{dashboardContent}</div>
              <ToastContainer />
            </div>
          </div>
        </div>
      </div>
    );
  }
}

Dashboard.propTypes = {
  profile: PropTypes.object.isRequired,
  getCurrentProfile: PropTypes.func.isRequired
};

const mapStateToProps = state => ({
  profile: state.profile,
  auth: state.auth
});

export default connect(
  mapStateToProps,
  { getCurrentProfile, deleteProfile, setErrorsEmpty, setLoggedIn }
)(Dashboard);
