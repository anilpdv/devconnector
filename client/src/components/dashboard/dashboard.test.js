import React from "react";
import { Dashboard } from "./dashboard";
import { shallow } from "enzyme";

describe("should render without crashing", () => {
  const getCurrentProfile = jest.fn();
  const props = {
    getCurrentProfile
  };
  const dashboard = shallow(<Dashboard {...props} />);

  it("<Dashborad/>", () => {
    expect(dashboard).toMatchSnapshot();
  });

  it("should call the `getCurrentProfile` func as th component mounts", () => {
    expect(getCurrentProfile).toHaveBeenCalled();
  });
});
