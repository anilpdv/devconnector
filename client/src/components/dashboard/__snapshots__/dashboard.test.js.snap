// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render without crashing <Dashborad/> 1`] = `ShallowWrapper {}`;
